src/StepAdaptationController/include/WalkingControllers/StepAdaptationController/StepAdaptationController.hpp

```diff
@@ -0,0 +1,272 @@
+        double m_remainingSingleSupportDuration;/**< The remained single support duration.*/
+        double m_omega;/**< The natural frequency of LIPM.*/
+
+        double m_currentTime;/**< The  current time.*/
+        double m_nextDoubleSupportDuration;/**< The timing of next double support.*/
+
+        int m_numberOfConstraint;  /**< Size of the input vector.. */
+
+        /**
+         *The buffered vectors for the interpolation of the foot trajectory
+         */
+        iDynTree::VectorDynSize m_xPositionsBuffer, m_yPositionsBuffer, m_zFirstPiecePositionsBuffer,m_zSecondPiecePositionsBuffer,
+                                m_yawsBuffer, m_timesBuffer, m_zFirstPieceTimesBuffer,m_zSecondPieceTimesBuffer;
+
+        iDynTree::ConvexHullProjectionConstraint m_convexHullComputer; /**< iDynTree convex hull helper. */
+        std::vector<iDynTree::Polygon> m_feetExtendedPolygon;/**< convex hull of the allowable landing foot position. */
+        iDynTree::Transform m_footTransform; /**< transform of the next foot position. */
+
+        bool m_isSolutionEvaluated{false}; /**< True if the solution is evaluated. */
+
+        /**
+         * Compute the hessian matrix.
+         * Please do not call this function to update the hessian matrix! It can be set only once.
+         * @return true/false in case of success/failure.
+         */
+        bool computeHessianMatrix();
+
+        /**
+         * Compute or update the linear constraints matrix(A) related to equality and inequality constraints(C<Ax<B)
+         * If the solver is already set the linear constraints matrix is updated otherwise it is set for
+         * the first time.
+         * @return true/false in case of success/failure.
+         */
+        bool computeConstraintsMatrix();
+
+        /**
+         * Compute or update the gradient
+         * @return true/false in case of success/failure.
+         */
+        bool computeGradientVector();
+
+        /**
+         * Set or update the lower and the upper bounds
+         * @return true/false in case of success/failure.
+         */
+        bool computeBoundsVectorOfConstraints();
+
+    public:
+        /**
+         * Constructor of step adaptation controller.
+         * @param inputSize size of the controlled input vector;
+         * @param numberOfAllConstraints number of equality and inequality constraints!
+         */
+        StepAdaptationController();
+
+        /**
+         * Get the state of the solver.
+         * @return true if the solver is initialized false otherwise.
+         */
+        bool isInitialized();
+
+        /**
+         * Initialize the solver.
+         * @return true/false in case of success/failure.
+         */
+        bool initialize();
+
+        /**
+         * Solve the optimization problem.
+         * @return true/false in case of success/failure.
+         */
+        bool solve();
```

Comment:
Why there are two ``solve`` methods?

Reply:
One of them was for `osqp` that generally was a confusing mistake everywhere in the code that I removed them!